server.py

```diff
@@ -38,14 +38,32 @@ def index():
 def list_products():
     category        = request.args.get('category')
     discontinued    = str2bool(request.args.get('discontinued'))
-    results     = []
+    min_Price       = request.args.get('minPrice')
+    max_Price       = request.args.get('maxPrice')
+    price           = request.args.get('price')
+    limit           = request.args.get('limit')
+    results         = []
+    count           = 0
+    if (limit is not None and int(limit) < 0):
+        return reply(results, HTTP_400_BAD_REQUEST)
+
     for key, value in products.iteritems():
-        if matchesClause(category,      value['category']) and \
-           matchesClause(discontinued,  value['discontinued']):
-             results.append(products[key])
+        if (limit is not None and int(count) == int(limit)):
+            break
+        if matches_clause(category,      value['category']) and \
```

Comment:
Thank you for changing the `matches_clause` method to "snake-case", as the professor had suggested :)